src/EditorFeatures/CSharpTest/ConvertTupleToStruct/ConvertTupleToStructTests.cs

```diff
@@ -3115,11 +3278,32 @@ void Goo()
     {
         var t1 = (a: 1, b: 2);
     }
-}
-        </Document>
-    </Project>
-</Workspace>";
-            await TestInRegularAndScriptAsync(text, expected, index: 3, options: this.PreferImplicitTypeWithInfo(), testHost: host);
+}";
+
+            var test = new VerifyCS.Test
+            {
+                CodeActionIndex = 3,
+                TestHost = host,
+                TestState =
+                {
+                    Sources = { text1 },
+                    AdditionalProjects =
+                    {
+                        ["DependencyProject"] = { Sources = { text2 } }
+                    },
+                },
+                FixedState =
+                {
+                    Sources = { expected1 },
+                    AdditionalProjects =
+                    {
+                        ["DependencyProject"] = { Sources = { expected2 } }
+                    },
+                },
+            };
+
+            test.Options.AddRange(PreferImplicitTypeWithInfo());
+            await test.RunAsync();
```

Comment:
Hard to tell what the action with index 3 means? What are the other offered actions?

Could this test use `ExactActionSetOffered` (or any other form that clarifies it a bit)?

Reply:
The recommended approach is always specifying _both_ `CodeActionIndex` and `CodeActionEquivalenceKey`. This form asserts that the action presented at a given index has a given equivalence key, which is both easy to read and strong assurance of intended order.

Reply:
ok.  can do so.